File: Minsk/CodeAnalysis/LiteralExpressionSyntax.cs
namespace Minsk.CodeAnalysis;

public class LiteralExpressionSyntax : ExpressionSyntax
{
    public SyntaxToken LiteralToken { get; }

    public override SyntaxKind Kind => SyntaxKind.LiteralExpression;
    public override IEnumerable<SyntaxNode> Children => new[] { LiteralToken };

    public LiteralExpressionSyntax(SyntaxToken literalToken)
    {
        LiteralToken = literalToken;
    }
}

File: nim/src/minsk/codeAnalysis/syntax/expressions/literalExpressionSyntax.nim
import minsk/codeAnalysis/syntax/[
  expressionSyntax,
  syntaxKind,
  syntaxNode,
  syntaxToken,
]

type
  LiteralExpressionSyntax* = ref object of ExpressionSyntax
    literalToken*: SyntaxToken

proc newLiteralExpressionSyntax*(literalToken: SyntaxToken): LiteralExpressionSyntax =
  new(result)
  result.literalToken = literalToken

method kind*(self: LiteralExpressionSyntax): SyntaxKind =
  return SyntaxKind.LiteralExpression

method children*(self: LiteralExpressionSyntax): seq[SyntaxNode] =
  result.add(self.literalToken)
